src/js/angular/clustermanagement/directives/cluster-legend.directive.js

```diff
@@ -0,0 +1,153 @@
+import {ClusterSvgService, HexagonConfig, LineConfig} from "../services/cluster-svg.service";
+
+const modules = [];
+angular.module('graphdb.framework.clustermanagement.directives.cluster-legend', modules)
+    .directive('clusterLegend', clusterLegend);
+
+clusterLegend.$inject = ['$rootScope', '$sce'];
+
+function clusterLegend($rootScope, $sce) {
+    return {
+        restrict: 'E',
+        templateUrl: 'js/angular/clustermanagement/templates/cluster-legend.html',
+        scope: {
+            clusterViewSelector: '='
+        },
+        link: ($scope, element, attrs) => {
+            // =========================
+            // Public variables
+            // =========================
+            $scope.nodeStates = [];
+            $scope.syncStatuses = [];
+            $scope.linkStates = [];
+
+            // =========================
+            // Private variables
+            // =========================
+            const legendElement = element.find('.cluster-legend-container');
+
+            // =========================
+            // Public functions
+            // =========================
+            $scope.toggleLegend = () => {
+                legendElement.toggleClass('hidden');
+            };
+
+            $scope.closeDialog = () => {
```

Comment:
It'd be better if there is a way to close the leged with the [esc] key.

Reply:
Done